.github/workflows/pythonpackage.yml

```diff
@@ -41,3 +44,4 @@ jobs:
           name: rich
           flags: unittests
           env_vars: OS,PYTHON
+          fail_ci_if_error: true
```

Comment:
By codecov errors, does it mean a missing line of coverage? I'd prefer not be be so strict. I'd prefer it to be a judgement call.

Reply:
This only means that the step will fail if there is an error in uploading the coverage data. In other words, your CI pipelines would fail if you removed the secret key or codecov servers went down. I'll remove it for now :)